src/overlays/actors/ovl_En_Part/z_en_part.c

```diff
@@ -21,23 +26,299 @@ const ActorInit En_Part_InitVars = {
+                }
+            }
+            func_8002F71C(globalCtx, this->actor.parent, (650.0f - this->actor.parent->xzDistFromLink) * 0.04f + 4.0f,
+                          this->actor.parent->posRot.rot.y, 8.0f);
+            player->invincibilityTimer = prevInvincibilityTimer;
+            this->timer = 1;
+        }
+
+        func_80033480(globalCtx, &this->actor.posRot.pos, 0.0f, 1, 300, 150, 1);
+        velocity.x = Math_Rand_CenteredFloat(16.0f);
+        EffectSsHahen_Spawn(globalCtx, &this->actor.posRot.pos, &velocity, &accel, 20,
+                            (s32)((Math_Rand_ZeroOne() * 5.0f + 12.0f) * 2), -1, 10, NULL);
+        Audio_PlayActorSound2(&this->actor, NA_SE_EN_MONBLIN_GNDWAVE - SFX_FLAG);
+    }
+}
+
+void func_80ACE7E8(EnPart* this, GlobalContext* globalCtx) {
+    Vec3f zero = { 0.0f, 0.0f, 0.0f };
+
+    if ((this->actor.parent == NULL) || (this->actor.parent->update == NULL)) {
+        EffectSsDeadDb_Spawn(globalCtx, &this->actor.posRot.pos, &zero, &zero, (s16)(this->actor.scale.y * 100.0f) * 40,
+                             7, 255, 255, 255, 255, 0, 255, 0, 1, 9, true);
+        Actor_Kill(&this->actor);
+        return;
+    }
+
+    if (this->timer == 0) {
+        f32 diffsSum =
+            Math_SmoothScaleMaxMinF(&this->actor.posRot.pos.x, this->actor.initPosRot.pos.x, 1.0f, 5.0f, 0.0f);
+        diffsSum += Math_SmoothScaleMaxMinF(&this->actor.posRot.pos.y, this->actor.initPosRot.pos.y, 1.0f, 5.0f, 0.0f);
+        diffsSum += Math_SmoothScaleMaxMinF(&this->actor.posRot.pos.z, this->actor.initPosRot.pos.z, 1.0f, 5.0f, 0.0f);
+        diffsSum += Math_SmoothScaleMaxMinF(&this->rotZ, 0.0f, 1.0f, 0.25f, 0.0f);
+        if (diffsSum == 0.0f) {
+            this->actor.parent->initPosRot.rot.x--;
+            this->timer--;
+            Audio_PlayActorSound2(&this->actor, NA_SE_EN_STAL_DAMAGE);
+        }
+    } else if (this->timer > 0) {
+        this->timer--;
+    }
+
+    if (this->actor.parent->colChkInfo.health != 0) {
+        Actor_Kill(&this->actor);
+    }
+}
+
+void EnPart_Update(Actor* thisx, GlobalContext* globalCtx) {
+    static EnPartActionFunc sActionFuncs[] = {
+        func_80ACDDE8, func_80ACE13C, func_80ACE5B8, func_80ACE5C8, func_80ACE7E8,
+    };
+
+    EnPart* this = THIS;
+
+    Actor_MoveForward(&this->actor);
+
+    if ((this->actor.params > 4 && this->actor.params < 9) || this->actor.params < 0) {
+        func_8002E4B4(globalCtx, &this->actor, 5.0f, 15.0f, 0.0f, 5);
+        if (this->actor.params >= 0) {
+            Math_SmoothScaleMaxMinF(&this->actor.speedXZ, 0.0f, 1.0f, 0.5f, 0.0f);
+            if (thisx->bgCheckFlags & 1) {
+                thisx->bgCheckFlags &= ~1;
+                thisx->velocity.y = 6.0f;
+            }
+        }
+    }
+
+    sActionFuncs[this->action](this, globalCtx);
+}
+
+Gfx* func_80ACEAC0(GraphicsContext* gfxCtx, u8 primR, u8 primG, u8 primB, u8 envR, u8 envG, u8 envB) {
+    Gfx* displayList;
+    Gfx* displayListHead;
```

Comment:
we've started going with dList and dList head as the abbreviations for these, but we havent made the change project wide yet. dont have to change it now if you dont want, but just letting you know